### DISCUSSION TITLE
Parameterization `OFFSET` value when an `ORDER BY` clause is used?

### DISCUSSION THREAD

**Discussion by amirrajan** [duckdb/duckdb#3719] [Q&A]
Is there a way for offset to be used as a parameter when an order by clause is used?

The following fails with an `Inconsistent types from for parameter with index` exception:
```js
await Duck.run(
      `
      select *
      from (
        values
          ('a', 1),
          ('b', 2),
          ('c', 3),
          ('d', 4)
        ) as T(ch, n)
      order by ch asc
      limit $1
      offset $2
      ;
    `,
      [2, 1]
    ).then((res) => {
      expect(res).toEqual([
        ['b', 2],
        ['c', 3],
      ]);
    });
 ```
 
 Removing the order by clause works:
```js
await Duck.run(
    `
    select *
    from (
      values
        ('a', 1),
        ('b', 2),
        ('c', 3),
        ('d', 4)
      ) as T(ch, n)
    limit $1
    offset $2
    ;
  `,
    [2, 1]
  ).then((res) => {
    expect(res).toEqual([
      ['b', 2],
      ['c', 3],
    ]);
  });
```

**Comment by Alex-Monahan** [comment]
Hmm, this may not help, but in some clients, it helps to put a cast expression after the parameter in the query, like `$2::int`.

**Reply by amirrajan** [reply]
Yea, I think it's a specific challenge with the nodejs interface. Not sure if there's a way to give a "type hint" of sorts on parameters that are passed in.
